### PR TITLE
Add support for Mesh Search Plugin and Reference Authentication plugin

### PR DESCRIPTION
**Pull request by netwarex**
Mesh Search Plugin, Reference Authentication Plugin support, Disable authentication, added did you mean and autocomplete to search demo, updated Mesh to 0.40.2

### CODE REVIEW

File: portal/app/Http/Kernel.php [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Kernel extends HttpKernel
     protected $middleware = [
         \App\Http\Middleware\CheckForMaintenanceMode::class,
         \Illuminate\Foundation\Http\Middleware\ValidatePostSize::class,
-        \App\Http\Middleware\Authenticate::class,
+        //\App\Http\Middleware\Authenticate::class,
```
Comment (yrucrem) [inline portal/app/Http/Kernel.php:19]: Is this intentional?

**Review by yrucrem — CHANGES_REQUESTED** [review]
Looks good to me.

File: portal/app/Http/Kernel.php (hunk 1 above)
Comment (netwarex) [reply portal/app/Http/Kernel.php:19]: Yes, authentication is disabled now.

Comment (berni2288) [reply portal/app/Http/Kernel.php:19]: Can we make authentication optional, e.g.: a login link?

File: README.demo.md [hunk 2]
```diff
@@ -35,6 +35,15 @@ If you would like to change the assets, please check the [Laravel Mix documentat
 
 ### Commercial Mesh Plugins
 
+#### Required for some demos
```
Comment (berni2288) [inline README.demo.md]: Please move this section before "### Running the portal" and add a link to https://getmesh.io/docs/plugin-system/#_deployment

**Review by berni2288 — CHANGES_REQUESTED** [review]
I think we need a topic in the README.demo.md that clearly says that all current docker volumes of `portal-php-reference` have to be deleted first before pulling a new version of portal-php-reference/demo with `git pull`. Also the Mesh API Key has to be removed from `.env` then. Also the composer dependencies have to be updated.

But if you want we can also create a new ticket for this topic and leave it out of this PR completely for now.

File: README.demo.md (hunk 2 above)
Comment (netwarex) [reply README.demo.md]: I have moved this to the "Running the portal" part

**Pull request merged**